Headers/AppKit/NSPasteboardItem.h

```diff
@@ -0,0 +1,81 @@
+/** <title>NSPasteboardItem</title>
+
+   <abstract>class for wrapping pasteboard content</abstract>
+
+   Copyright <copy>(C) 2017 Free Software Foundation, Inc.</copy>
+
+   Author: Daniel Ferreira <dtf@stanford.edu>
+   Date: July 2017
+
+   This file is part of the GNUstep GUI Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with this library; see the file COPYING.LIB.
+   If not, write to the Free Software Foundation,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifndef _GNUstep_H_NSPasteboardItem
+#define _GNUstep_H_NSPasteboardItem 
+
+#import <AppKit/AppKitDefines.h>
+#import <AppKit/NSPasteboard.h>
+#import <Foundation/NSArray.h>
+#import <Foundation/NSDictionary.h>
+#import <CoreFoundation/CFBase.h>
+
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_6, GS_API_LATEST)
+
+@protocol NSPasteboardItemDataProvider;
+
+@interface NSPasteboardItem : NSObject <NSPasteboardWriting, NSPasteboardReading> {
+  NSMutableDictionary *_providerMap;
+  NSMutableDictionary *_dataMap;
+  NSMutableArray *_types;
+}
+#if GS_HAS_DECLARED_PROPERTIES
+@property (readonly, copy) NSArray *types;
```

Comment:
This declaration comes from the documentation and I assume that this `copy` is not there by accident. I implemented the getter with copying.

I'm not sure we need `nonatomic` here. Despite us having a custom getter, I do not see it as thread-unsafe.